plugins/BEdita/Core/tests/TestCase/Command/ObjectsDeleteCommandTest.php

```diff
@@ -0,0 +1,125 @@
+ */
+class ObjectsDeleteCommandTest extends TestCase
+{
+    use ConsoleIntegrationTestTrait;
+
+    /**
+     * Fixtures
+     *
+     * @var array
+     */
+    protected $fixtures = [
+        'plugin.BEdita/Core.ObjectTypes',
+        'plugin.BEdita/Core.PropertyTypes',
+        'plugin.BEdita/Core.Properties',
+        'plugin.BEdita/Core.Relations',
+        'plugin.BEdita/Core.RelationTypes',
+        'plugin.BEdita/Core.Objects',
+        'plugin.BEdita/Core.Locations',
+        'plugin.BEdita/Core.Media',
+        'plugin.BEdita/Core.Profiles',
+        'plugin.BEdita/Core.Users',
+        'plugin.BEdita/Core.Streams',
+    ];
+
+    /**
+     * @inheritDoc
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->useCommandRunner();
+    }
+
+    /**
+     * Test buildOptionParser method
+     *
+     * @return void
+     * @covers ::buildOptionParser()
+     */
+    public function testBuildOptionParser()
+    {
+        $this->exec('objects_delete --help');
+        $this->assertOutputContains('Delete objects in trash since this date');
+        $this->assertOutputContains('Delete objects in trash by type');
+    }
+
+    /**
+     * Test `execute` method
+     *
+     * @return void
+     * @covers ::execute()
+     * @covers ::objectsIterator()
+     * @covers ::deleteObject()
+     */
+    public function testExecute(): void
+    {
+        $this->exec('objects_delete --type documents');
+        $this->assertOutputContains('Deleting from trash objects, since -1 month, for type(s) documents');
+        $this->assertOutputContains('Deleted from trash 2 objects [0 errors]');
+        $this->assertOutputContains('Done');
+        $this->assertExitSuccess();
+    }
+
+    /**
+     * Test `execute` method
+     *
+     * @return void
+     * @covers ::execute()
+     * @covers ::objectsIterator()
+     * @covers ::deleteObject()
+     */
+    public function testExecuteSince(): void
+    {
+        $this->exec('objects_delete --since "-1 weeks"');
+        $this->assertOutputContains('Deleting from trash objects, since -1 weeks');
+        $this->assertOutputContains('Deleted from trash 2 objects [0 errors]');
+        $this->assertOutputContains('Done');
+        $this->assertExitSuccess();
+    }
+
+    /**
+     * Test `deleteObject` method
+     *
+     * @return void
+     * @covers ::deleteObject()
+     */
+    public function testDeleteObject(): void
+    {
+        $deleted = $errors = 0;
+        $io = $this->getMockBuilder('Cake\Console\ConsoleIo')->getMock();
+        $object = new ObjectEntity([]);
+        $command = new ObjectsDeleteCommand();
+        $reflectionClass = new \ReflectionClass($command);
+        $method = $reflectionClass->getMethod('deleteObject');
+        $method->setAccessible(true);
+        $method->invokeArgs($command, [$io, $object, &$deleted, &$errors]);
+        $this->assertEquals(0, $deleted);
+        $this->assertEquals(1, $errors);
```

Comment:
In general instead of change the method visibility to test it, I think it would be better testing it throught the expected flow.

So, for example, you could count trash items before run the command and count items after the execution testing the expected result in different situations (using `since`, `type` options, ...)

Reply:
good point. in this case it's difficult to test it because the exception cannot be raised, unless you pass a wrong entity to the private method
```php
foreach ($this->objectsIterator($objectType, $conditions) as $object) {
    $this->deleteObject($io, $object, $deleted, $errors);
}
```
This code won't raise exception, because `objectsIterator` always return entities that can be deleted.

Reply:
You could try to reach that condition with a listener on 'Model.beforeDelete`.
Something as

```php
$throwError = function () {
    throw new \Exception('Fake error');
};
// add listener to global event manager
EventManager::instance()->on('Model.beforeDelete', $throwError);

// count trash items

// run command

// ensure to off listener from global event manager
EventManager::instance()->off('Model.beforeDelete', $throwError);

// count new trash items

// test assertion
```

Reply:
done